lib/sensu/server/process.rb

```diff
@@ -207,21 +207,19 @@ def aggregate_check_result(result)
       #
       # @param client [Hash]
       # @param check [Hash]
-      # @param result [Hash]
       # @param callback [Proc] to call when the check result data has
       #   been stored (history, etc).
-      def store_check_result(client, check, result, &callback)
-        @logger.debug("updating result data", :check=> check)
-        result_truncated_output = result.dup
-        result_truncated_output[:check][:output] = check[:output][0..256]
-        @redis.set("result:#{client[:name]}:#{check[:name]}", MultiJson.dump(result_truncated_output))
-        @redis.sadd('results', "#{client[:name]}:#{check[:name]}")
-        @redis.sadd("history:#{client[:name]}", check[:name])
+      def store_check_result(client, check, &callback)
+        @logger.debug("storing check result", :check => check)
+        @redis.sadd("result:#{client[:name]}", check[:name])
         result_key = "#{client[:name]}:#{check[:name]}"
-        history_key = "history:#{result_key}"
-        @redis.rpush(history_key, check[:status]) do
-          @redis.ltrim(history_key, -21, -1)
-          callback.call
+        check_truncated = check.merge(:output => check[:output][0..256])
```

Comment:
merge() will create a new object, effectively doing the same as dup() with assignment.